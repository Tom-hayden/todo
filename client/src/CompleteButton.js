import React from "react";
import {completeTodo} from "./buttonFunctions";


const CompleteButton = ({todo, socket}) => {
    return (
        <button onClick={() => completeTodo(socket, todo)} id={"complete_" + todo.id}>
            Complete
        </button>
    )
}

export default CompleteButton;