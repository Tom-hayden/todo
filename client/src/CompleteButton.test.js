import CompleteButton from "./CompleteButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
    const todoID = "3";
    const todo = {
        id: todoID
    };
    const socket = {};
    const component = renderer.create(
        CompleteButton({todo, socket})
    ).toJSON();

    expect(component).toMatchSnapshot();
});
