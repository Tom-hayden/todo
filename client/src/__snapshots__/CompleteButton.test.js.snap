// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ui compact button"
  id="complete_3"
  onClick={[Function]}
>
  Complete
</button>
`;
